core/trino-main/src/test/java/io/trino/sql/query/TestGroupBy.java

```diff
@@ -253,5 +253,80 @@ SELECT sum(a)
                 GROUP BY DISTINCT AUTO
                 """))
                 .matches("VALUES (BIGINT '2')");
+
+        // ROLLUP
```

Comment:
Add at least one test that produces mixed groups from AUTO vs the other grouping sets. For instance:

```
SELECT a, b, c, sum(d)
FROM ...
GROUP BY AUTO, ROLLUP(a)
```

It should result in the following grouping sets: `(), (a), (a, b, c)`

Reply:
Right, when there are multiple grouping set specifications, their grouping sets are cross joined. I forgot about that. So, in this case, it would be `((a, b, c)) x ((), (a))`, which results in `((a, b, c), (a, b, c))`. That's unfortunate, as it makes `AUTO` not that useful when combined with other grouping sets.

Reply:
Added the tests anyway. Let me know if we want to disallow `AUTO` with other grouping sets.

Reply:
Let's leave it as is for consistency with other grouping sets. AUTO is just a short hand for a grouping set formed by all the non-aggregate columns.